x-pack/test/api_integration/apis/slos/get_slo.ts

```diff
@@ -0,0 +1,456 @@
+          version: 2,
+          instanceId: '*',
+          summary: {
+            sliValue: 0.5,
+            errorBudget: {
+              initial: 0.01,
+              consumed: 50,
+              remaining: -49,
+              isEstimated: false,
+            },
+            status: 'VIOLATED',
+          },
+        });
+      });
+    });
+
+    it('gets slo by id and calculates SLI - timeslices calendarAligned', async () => {
+      const id = await createSLO({
+        groupBy: '*',
+        timeWindow: {
+          duration: '1w',
+          type: 'calendarAligned',
+        },
+        budgetingMethod: 'timeslices',
+        objective: {
+          target: 0.99,
+          timesliceTarget: 0.95,
+          timesliceWindow: '10m',
+        },
+      });
+
+      await retry.tryForTime(300 * 1000, async () => {
+        const getResponse = await supertestAPI
+          .get(`/api/observability/slos/${id}`)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+
+        expect(getResponse.body).eql({
+          name: 'Test SLO for api integration',
+          description: 'Fixture for api integration tests',
+          indicator: {
+            type: 'sli.kql.custom',
+            params: {
+              index: 'kbn-data-forge*',
+              filter: `system.network.name: eth1`,
+              good: 'container.cpu.user.pct < 1',
+              total: 'container.cpu.user.pct: *',
+              timestampField: '@timestamp',
+            },
+          },
+          budgetingMethod: 'timeslices',
+          timeWindow: { duration: '1w', type: 'calendarAligned' },
+          objective: {
+            target: 0.99,
+            timesliceTarget: 0.95,
+            timesliceWindow: '10m',
+          },
+          tags: ['test'],
+          groupBy: '*',
+          id,
+          settings: { syncDelay: '1m', frequency: '1m' },
+          revision: 1,
+          enabled: true,
+          createdAt: getResponse.body.createdAt,
+          updatedAt: getResponse.body.updatedAt,
+          version: 2,
+          instanceId: '*',
+          summary: {
+            sliValue: 0,
+            errorBudget: {
+              initial: 0.01,
+              consumed: 0.198413,
+              remaining: 0.801587,
+              isEstimated: false,
+            },
+            status: 'DEGRADING',
+          },
+        });
+      });
+    });
+
+    it('gets slos by query', async () => {
+      const id = await createSLO();
+
+      await retry.tryForTime(300 * 1000, async () => {
+        const response = await supertestAPI
+          .get(`/api/observability/slos`)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+
+        expect(response.body.results.length).eql(2);
+
+        const searchResponse = await supertestAPI
+          .get(`/api/observability/slos?kqlQuery=slo.name%3A+api*`)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
```

Comment:
Could you also test without the `slo.name` but directly using the name of the SLO (or part of it using *), this should rely on the full text search